app/src/main/res/layout/fragment_city_list.xml

```diff
@@ -5,6 +5,36 @@
     android:layout_height="match_parent"
     android:background="#2D71B6">
 
+    <androidx.constraintlayout.widget.ConstraintLayout
+        xmlns:android="http://schemas.android.com/apk/res/android"
+        xmlns:app="http://schemas.android.com/apk/res-auto"
+        xmlns:tools="http://schemas.android.com/tools"
+        android:id="@+id/stateLayout"
+        android:orientation="horizontal"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:gravity="center_vertical"
+        >
+        <ImageView
```

Comment:
You should separate the loading/error images into their own elements so it's easier to toggle their visibility. Also, instead of using an imageView for the loading indicator, you should be able to use
`<com.google.android.material.progressindicator.CircularProgressIndicator`